src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -1514,6 +1514,7 @@ because xbuild doesn't support framework reference assemblies.
   </ItemGroup>
 
   <Touch Files="$(_AndroidStampDirectory)_GenerateJavaStubs.stamp" AlwaysCreate="True" />
+  <Touch Files="@(_TypeMapAssemblySource)" />
```

Comment:
Alternatively, do we know what deleted `@(_TypeMapAssemblySource)`? Are they missing from `FileWrites`?

Reply:
That's the problem, we have no idea why they are gone. The guess is that some earlier build deleted them (DTB perhaps?) which we don't have build logs for.

Reply:
Regardless of what made the files disappear, I don't think a stamp file in `Outputs` is enough, at least in this case.  The task inside the target produces files, and we must make sure that those files exist (whether they were removed by us for some reason, or by the developer) and so they must be part of the outputs.

Reply:
So I wonder if it is this. 

Because the `_TypeMapAssemblySource` are generated via a Task. If the task does not run, the `_TypeMapAssemblySource` will be empty. 

On a DTB is `_GeneratedJavaStubs` is skipped the task filling in the `_TypeMapAssemblySource` ItemGroup will be empty. As a result the `FileWrites` entry will not include the files. 

In the case where the Task is not run, we need a way to populate the `_TypeMapAssemblySource` from the files on disk to make sure they are not deleted.

Reply:
We have this chain of dependencies:

```xml
<Target Name="_PrepareNativeAssemblySources">
  <PrepareAbiItems
      BuildTargetAbis="@(_BuildTargetAbis)"
      NativeSourcesDir="$(_NativeAssemblySourceDir)"
      InstantRunEnabled="$(_InstantRunEnabled)"
      Debug="$(AndroidIncludeDebugSymbols)"
      Mode="typemap">
    <Output TaskParameter="AssemblySources" ItemName="_TypeMapAssemblySource" />
    <Output TaskParameter="AssemblyIncludes" ItemName="_TypeMapAssemblyInclude" />
  </PrepareAbiItems>
</Target>

<PropertyGroup>
  <_GenerateJavaStubsDependsOnTargets>
    _SetLatestTargetFrameworkVersion;
    _PrepareAssemblies;
    _PrepareNativeAssemblySources;
    $(_AfterPrepareAssemblies);
  </_GenerateJavaStubsDependsOnTargets>
</PropertyGroup>

<Target Name="_GenerateJavaStubs"
    DependsOnTargets="$(_GenerateJavaStubsDependsOnTargets);$(BeforeGenerateAndroidManifest)"
    Inputs="@(_AndroidMSBuildAllProjects);@(_ResolvedUserMonoAndroidAssemblies);$(_AndroidManifestAbs);$(_AndroidBuildPropertiesCache);@(AndroidEnvironment);@(LibraryEnvironments)"
    Outputs="$(_AndroidStampDirectory)_GenerateJavaStubs.stamp">
```

Will `_PrepareNativeAssemblySources` be ignored when `_GenerateJavaStubs` `Outputs` are deemed up to date?